src/main.rs

```diff
@@ -150,6 +169,78 @@ fn get_stdin_data() -> Result<String> {
     Ok(input)
 }
 
+/// Target object for which the trait must be derived.
+#[derive(Debug, Clone, PartialEq)]
+enum DeriveTarget {
+    /// Derive the trait for all types
+    All,
+    /// Derive the trait for a specific type only.
+    Type(String),
+    /// Derive the trait for all structs.
+    Structs,
+    /// Derive the trait for enums, optionally only for simple
+    /// ([unit-only](https://doc.rust-lang.org/reference/items/enumerations.html)) enums.
+    Enums {
+        /// Limit trait derivation to *unit-only* enums.
+        unit_only: bool,
+    },
+}
+
+/// A trait to derive, as well as the object for which to derive it.
+#[derive(Debug, Clone, PartialEq)]
+struct Derive {
+    /// Target object (type, structs, enums) to derive the trait for.
+    pub target: DeriveTarget,
+    /// Trait to derive for the target.
+    pub derived_trait: String,
+}
+
+impl Derive {
+    pub fn all(derived_trait: &str) -> Self {
+        Derive {
+            target: DeriveTarget::All,
+            derived_trait: derived_trait.to_owned(),
+        }
+    }
+}
+
+fn parse_derive(arg: &str) -> Result<Derive> {
```

Comment:
this could be an `impl FromStr for Derive`

Reply:
Well spotted, should be an easy fix.